lib/sync/worker.js

```diff
@@ -0,0 +1,59 @@
+var util = require('util');
+var syncUtil = require('./util');
+var LOGGER_NAME = syncUtil.SYNC_LOGGER;
+
+/**
+ * @function QueueWorker~processor
+ * @param {Object} job the job from the queue.
+ * @param {String} job.id the mongodb id of the job
+ * @param {String} job.ack a unique id of the message in order to ack
+ * @param {Object} job.payload the payload of the message
+ * @param {Number} job.tries the number of tries of the job
+ * @param {Function} callback the callback function that should be invoked when the job is completed
+ */
+
+/**
+ * construct a new worker that will consume the jobs from the given queue and process it with the given processor.
+ * @param {MongodbQueue} queue an instance of the MongodbQueue
+ * @param {QueueWorker~processor} the job processor
+ * @param {Object} opts configuration options
+ * @param {Number} opts.interval the interval between job runs
+ */
+function QueueWorker(queue, processor, opts) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Rename this 'QueueWorker' function to match the regular expression ^[a-z][a-zA-Z0-9]*$ [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS100)